library/src/main/java/com/google/android/exoplayer2/text/eia608/Eia608Decoder.java

```diff
@@ -463,19 +719,19 @@ private String getDisplayCaption() {
 
     int endIndex = endsWithNewline ? buildLength - 1 : buildLength;
     if (captionMode != CC_MODE_ROLL_UP) {
-      return captionStringBuilder.substring(0, endIndex);
+      return captionStringBuilder.subSequence(0, endIndex);
     }
 
     int startIndex = 0;
     int searchBackwardFromIndex = endIndex;
     for (int i = 0; i < captionRowCount && searchBackwardFromIndex != -1; i++) {
-      searchBackwardFromIndex = captionStringBuilder.lastIndexOf("\n", searchBackwardFromIndex - 1);
+      searchBackwardFromIndex = captionStringBuilder.toString().lastIndexOf("\n", searchBackwardFromIndex - 1);
```

Comment:
This entire logic can be removed as soon as every line is implemented with a separate Cue object, but until then:
what is the problem with the lastIndexOf() of StringBuilder? Why do you use the String one?
https://docs.oracle.com/javase/7/docs/api/java/lang/StringBuilder.html#lastIndexOf(java.lang.String,%20int)